disnake/ui/item.py

```diff
@@ -142,7 +145,7 @@ def view(self) -> V_co:
         """Optional[:class:`View`]: The underlying view for this item."""
         return self._view
 
-    async def callback(self, interaction: MessageInteraction, /) -> None:
+    async def callback(self, interaction: MessageInteraction[ClientT], /) -> None:
```

Comment:
`Modal.callback` and `Modal.on_error` should also have this `[ClientT]`.
There are a few more places with unknown generic parameters, but those should all be are internal and don't matter too much for now.

Reply:
https://github.com/DisnakeDev/disnake/pull/1037/commits/56eee7831d1d80457bd6bc4f134fbed66385f61e